magento/pwa-studio
# Add Adobe client data layer

## Description

Add the Adobe Client Data Layer as an extension, enabling product recommendations and others to do their work. As an extension, the script is optional, and it should be loaded with the [lowest priority possible](https://addyosmani.com/blog/script-priorities/). Webpack's [dynamic imports with magic comments](https://webpack.js.org/api/module-methods/#magic-comments) make this pretty straightforward.

## Related Issue

## Acceptance

### Verification Stakeholders

### Specification

### Verification Steps

1. Verify that `window.adobeDataLayer` eventually evaluates to an object.
2. Verify that the library isn't included in any of our bundles and doesn't delay rendering.

## Screenshots / Screen Captures (if appropriate)

## Checklist
* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.


### Resolved issues:
1. [x] resolves magento/pwa-studio#2886: Add Adobe client data layer
